cachi2/core/models/sbom.py

```diff
@@ -93,3 +96,401 @@ class Sbom(pydantic.BaseModel):
+    """SPDX Package External Reference for category package-manager.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: Literal["PACKAGE-MANAGER"]
+
+
+class SPDXPackageExternalRefPackageManagerMaven(pydantic.BaseModel):
+    """SPDX Package External Reference for category package-manager and type maven-central.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: Literal["PACKAGE-MANAGER"]
+    referenceLocator: str = pydantic.Field(pattern=r"[^:]+:[^:]+(:[^:]+)?")
+    referenceType: Literal["maven-central"]
+
+
+class SPDXPackageExternalRefPackageManagerNPM(SPDXPackageExternalRefPackageManager):
+    """SPDX Package External Reference for category package-manager and type npm.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: Literal["PACKAGE-MANAGER"]
+    referenceLocator: str = pydantic.Field(pattern=r"^[^@]+@[^@]+$")
+    referenceType: Literal["npm"]
+
+
+class SPDXPackageExternalRefPackageManagerNuget(SPDXPackageExternalRefPackageManager):
+    """SPDX Package External Reference for category package-manager and type nuget.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: Literal["PACKAGE-MANAGER"]
+    referenceLocator: str = pydantic.Field(pattern=r"^[^\/]+\/[^\/]+$")
+    referenceType: Literal["nuget"]
+
+
+class SPDXPackageExternalRefPackageManagerBower(SPDXPackageExternalRefPackageManager):
+    """SPDX Package External Reference for category package-manager and type bower.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: Literal["PACKAGE-MANAGER"]
+    referenceLocator: str = pydantic.Field(pattern=r"^[^#]+#[^#]+$")
+    referenceType: Literal["bower"]
+
+
+class SPDXPackageExternalRefPackageManagerPURL(
+    SPDXPackageExternalRefPackageManager, SPDXPackageExternalRefReferenceLocatorURI
+):
+    """SPDX Package External Reference for category package-manager and type purl.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: Literal["PACKAGE-MANAGER"]
+    referenceType: Literal["purl"]
+
+
+SPDXPackageExternalRefPackageManagerType = Annotated[
+    Union[
+        SPDXPackageExternalRefPackageManagerMaven,
+        SPDXPackageExternalRefPackageManagerNPM,
+        SPDXPackageExternalRefPackageManagerNuget,
+        SPDXPackageExternalRefPackageManagerBower,
+        SPDXPackageExternalRefPackageManagerPURL,
```

Comment:
Why do we have Maven, Nuget and Bower package managers, when cachi2 doesn't support them?
How about pip, go, bundler?

Reply:
or what's the purpose of all those classes, when we use only purl?

Reply:
> Why do we have Maven, Nuget and Bower package managers, when cachi2 doesn't support them?
How about pip, go, bundler?
Those are not supported. See https://spdx.github.io/spdx-spec/v2.3/external-repository-identifiers/. pypi is usually reference by purl like pkg:pypi/chardet@3.0.4 - at least that's what I saw everytime I tried to run syft.
So you think cachi2 should only support purl for PACKAGE-MANAGER? 

> or what's the purpose of all those classes, when we use only purl?
To make it standard complete - at least that field as we need it for purl.

Reply:
ok, if this makes parser complete, I'm fine with current implementation

Reply:
> So you think cachi2 should only support purl for PACKAGE-MANAGER?

Yes, because otherwise it looks like you're trying to turn cachi2 into something it is not - a SPDX tool.

Reply:
Closing this since it was addressed